snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -569,8 +570,11 @@ def mark_build_done(self):
             scriptlet_metadata=self._scriptlet_metadata[steps.BUILD]))
 
     def _get_machine_manifest(self):
+        # Use subprocess directly here. common.run_output will use binaries out
+        # of the snap, and we want to use the one on the host.
+        uname = subprocess.check_output(['uname', '-srvmpio'])
         return {
-            'uname': common.run_output(['uname', '-srvmpio']),
+            'uname': uname.decode(sys.getfilesystemencoding()).strip(),
```

Comment:
We will also need to catch unicode decode errors and use a surrogate on errors.

Reply:
Not sure how uname would cause this, but seems harmless. I added the extra case.